front/lib/resources/data_source_resource.ts

```diff
@@ -212,6 +212,16 @@ export class DataSourceResource extends BaseResource<DataSource> {
     };
   }
 
+  isManaged(): boolean {
+    return (
+      this.name.startsWith("managed-") &&
```

Comment:
I've kept the check on the name to account for the "managed" aspect, and since we already have some logic dependent on this check, I prefer to keep it.

Reply:
FYI I think we have a flow (second managed data source for slack as an example) that can allow selected users to create managed data sources with a custom name. We push them to use the managed prefix but I thiiink it's possible to create a managed data source of arbitrary name with the code today.